Core/Core/TextSubmission/TextSubmissionViewController.swift

```diff
@@ -0,0 +1,153 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import UIKit
+
+public final class TextSubmissionViewController: UIViewController, ErrorViewController, RichContentEditorDelegate {
+    // MARK: - Outlets
+
+    @IBOutlet private weak var contentView: UIView!
+    @IBOutlet private weak var keyboardSpace: NSLayoutConstraint!
+
+    // MARK: - Properties
+    public var editor: RichContentEditorViewController!
+    public var keyboard: KeyboardTransitioning?
+    private var assignmentID: String!
+    private var courseID: String!
+    private var env = AppEnvironment.shared
+    private var userID: String!
+    private var htmlContent = ""
+    private let isHorizonApp = AppEnvironment.shared.app != .student
+
+    // MARK: - Public Properties
+    public var didSetHtmlContent: ((String) -> Void) = { _ in}
+
+    public static func create(env: AppEnvironment, courseID: String, assignmentID: String, userID: String) -> TextSubmissionViewController {
+        let controller = loadFromStoryboard()
+        controller.assignmentID = assignmentID
+        controller.courseID = courseID
+        controller.userID = userID
+        controller.env = env
+        controller.editor = RichContentEditorViewController
+            .create(env: env, context: .course(courseID), uploadTo: .myFiles)
+        return controller
+    }
+
+    public static func create(htmlContent: String, courseID: String) -> TextSubmissionViewController {
```

Comment:
`assignmentID` and `userID` are not set in this `create()` method, but those are force unwrapped properties.
I see that we are not using them, because the "Done" button doesn't call the `submit()` method, but this seems very fragile.
Could you please make those properties optional properly and guard them in the submit method?